app/models/concerns/operations/stock_operations.rb

```diff
@@ -29,6 +29,56 @@ def register_loss(package, quantity:, from_location:)
       )
     end
 
+    def pack_or_unpack(params, user_id)
+      PackUnpack.new(params, user_id).run
+    end
+
+    class PackUnpack
+      def initialize(params, user_id)
+        @cause = Package.find(params[:id]) # box or pallet
+        @item = Package.find(params[:item_id]) # item to add or remove
+        @action = params[:task] # action pack or unpack
+        @user_id = user_id
+        @quantity = params[:quantity] # quantity to pack or unpack
+      end
+
+      def run
+        pkg_inventory = register_change
+        if pkg_inventory&.save
+          return { packages_inventory: pkg_inventory, success: true }
+        elsif pkg_inventory&.errors
+          return { errors: pkg_inventory.errors.full_messages, success: false }
+        end
+      end
+
+      private
+
+      def register_change
+        return unless action
```

Comment:
I'd personally be in favour of throwing an error in this case. Otherwise a bad action might go unnoticed